File: pwsdbcreator/pwsdb/src/main/java/com/alelk/pws/pwsdb/table/PwsBookStatisticTable.java
package com.alelk.pws.pwsdb.table;

import android.database.sqlite.SQLiteDatabase;

/**
 * Created by Alex Elkin on 17.03.2016.
 */
public class PwsBookStatisticTable implements PwsTable {
    public static final String TABLE_BOOKSTATISTIC = "bookstatistic";
    public static final String COLUMN_ID = "_id";
    public static final String COLUMN_BOOKID = "bookid";
    public static final String COLUMN_USERPREFERENCE = "userpref";
    public static final String COLUMN_READINGS = "readings";
    public static final String COLUMN_RATING = "rating";

    private static final String TABLE_CREATE_SCRIPT = "create table " + TABLE_BOOKSTATISTIC +
            "(" + COLUMN_ID + " integer primary key autoincrement, " +
            COLUMN_BOOKID + " integer not null, " +
            COLUMN_USERPREFERENCE + " integer, " +
            COLUMN_READINGS + " integer, " +
            COLUMN_RATING + " integer, " +
            "FOREIGN KEY (" + COLUMN_BOOKID + ") " +
            "REFERENCES " + PwsBookTable.TABLE_BOOKS + " (" +
            PwsBookTable.COLUMN_ID + "));";

    private static final String TABLE_DROP_SCRIPT = "drop table if exists " + TABLE_BOOKSTATISTIC;

    public static void createTable(SQLiteDatabase db) {
        db.execSQL(TABLE_CREATE_SCRIPT);
    }

    public static void dropTable(SQLiteDatabase db) {
        db.execSQL(TABLE_DROP_SCRIPT);
    }
}

File: pws/database/src/main/kotlin/com/alelk/pws/database/table/PwsBookStatisticTable.kt
/*
 * Copyright (C) 2018 The P&W Songs Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.alelk.pws.database.table

/**
 * Pws Book Statistic Table
 *
 * Created by Alex Elkin on 17.03.2016.
 */
object PwsBookStatisticTable : PwsTable {
    const val TABLE_BOOKSTATISTIC = "bookstatistic"
    const val COLUMN_ID = "_id"
    const val COLUMN_BOOKID = "bookid"
    const val COLUMN_USERPREFERENCE = "userpref"
    const val COLUMN_READINGS = "readings"
    const val COLUMN_RATING = "rating"
}